lib/virtio/virtqueue.c

```diff
@@ -632,6 +733,13 @@ static int vq_ring_must_notify(struct virtqueue *vq)
 {
 	uint16_t new_idx, prev_idx, event_idx;
 
+#ifdef VIRTIO_D_CACHE
+	metal_cache_invalidate(VRING_USED(vq), vring_used_size(VRING_NUM(vq)));
```

Comment:
only slave should flush this

Reply:
We need to invalidate (no flush right?) also for master because used could have been modified by slave in `vq_ring_enable_interrupt()`

Reply:
Sorry i was spoken about invalidate yes.
vq_ring_must_notify is called by virtqueue_kick at the end of a message sent or received, so caches has already been invalidated, i wonder if we could supress oinvalidate